File: frontend/src/app/models/loan.ts
import { BookCopy } from './copy';
import { Member } from './member';
export type LoanStatus = 'ALL' | 'ACTIVE' | 'OVERDUE';
export interface Loan {
  id: number;
  bookCopy: BookCopy;
  member: Member;
  checkoutDate: string;
  dueDate: string;
  returnDate: string | null;
  issuedBy: string;
  overdue: boolean;
}

File: backend/src/main/java/dev/yagiz/kulliyat/entity/Loan.java
package dev.yagiz.kulliyat.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDate;

@Entity
@Table(name = "loans")
@Getter
@Setter
public class Loan {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "book_copy_id", nullable = false)
    private BookCopy bookCopy;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    @Column(nullable = false)
    private LocalDate checkoutDate;

    @Column(nullable = false)
    private LocalDate dueDate;

    // Burası null'sa kitap şu an ödünç alınmış yani kütüphanede değil
    private LocalDate returnDate;

    // Kitabı ödünç veren Staff'ın kullanıcı adı
    @Column(nullable = false)
    private String issuedBy;

}
